### PR TITLE
GitHub 978

### LINKED ISSUE #978: Improve null checks and code strong typing
### Description

Most null pointer issues just result in general reliability issues, but if an attacker can intentionally trigger a null pointer dereference, they may be able to use the resulting exception to bypass security logic or to cause the application to reveal debugging information that will be valuable in planning subsequent attacks.

In some places, else if statments are either not correctly using null check, or are in the wrong order. In `compose-mail.component.ts`, for example- `else if (this.selectedMailbox.signature) {...} (line 1304)` comes before` else if (this.selectedMailbox) {...} (line 1310)`, so the null check on the `selectedMailbox` property is effectively happening before you are using it. And in `mail.redusers.ts`, the `<any>new Date(previous_updated)` takes a parameter that is defined as: `const previous_updated = previous.updated || null`, meaning that `null` could be passed to `Date()` which would not be valid (it expected either/ only a number or string value). Another example, in compose-mail.component.ts where the null check happens after the variable is already used - line 108 contains `if (value) {...}`, but above on line 103 you've used `if (value.url) {...}`.

CTemplar uses TypeScript, but it not making use of it's static type checking features. Again, this is less of a security issue, and more of a software stability and code quality problem. There are literally hundreds of instances of any and none of them are necessary (e.g. in `mail-detail.component.ts`). Consider enabling `noImplicitAny` in your `tsconfig.json` and enforcing it before code can be committed.

### PR DESCRIPTION
**Pull request by happyman125**
Fixes #978 

### Changes description

  - typescript checking and null check

### CODE REVIEW

**Comment by happyman125** [comment]
@The-Hidden-Hand I didn't complete all null check because it will take more time. but completed code strong typing. So please review and let me know

File: src/app/store/services/users.service.ts [hunk 1]
```diff
@@ -253,7 +253,7 @@ export class UsersService {
     return this.http.get<any>(url, body);
   }
 
-  addWhiteList(email, name) {
+  addWhiteList(email: any, name: any) {
```
Comment (The-Hidden-Hand) [inline src/app/store/services/users.service.ts:256]: Shouldn't be string types?

File: src/app/store/services/users.service.ts [hunk 2]
```diff
@@ -239,7 +239,7 @@ export class UsersService {
     return authenticated;
   }
 
-  getAccounts(id) {
+  getAccounts(id: string) {
```
Comment (The-Hidden-Hand) [inline src/app/store/services/users.service.ts:242]: Shouldn't be a number type?

File: src/app/store/services/users.service.ts [hunk 3]
```diff
@@ -275,13 +275,13 @@ export class UsersService {
...
 
-  addBlackList(email, name) {
+  addBlackList(email: any, name: any) {
     const url = `${apiUrl}users/blacklist/`;
     const body = { email, name };
     return this.http.post<any>(url, body);
   }
 
-  deleteBlackList(id) {
+  deleteBlackList(id: any) {
```
Comment (The-Hidden-Hand) [inline src/app/store/services/users.service.ts:284]: Shouldn't be a number type?

File: src/app/store/services/users.service.ts [hunk 4]
```diff
@@ -275,13 +275,13 @@ export class UsersService {
     return this.http.get<any>(url, body);
   }
 
-  addBlackList(email, name) {
+  addBlackList(email: any, name: any) {
```
Comment (The-Hidden-Hand) [inline src/app/store/services/users.service.ts:278]: Shouldn't be string types?

**Review by The-Hidden-Hand — CHANGES_REQUESTED** [review]
Please review and ensure accurate types.

**Comment by The-Hidden-Hand** [comment]
@happyman125 
There are many types not properly defined. Modals should be `NgbModalRef`, and so on.

**Comment by The-Hidden-Hand** [comment]
I may merge this PR after this week's release, and we do another iteration later keeping the issue open.

**Pull request merged**